src/gtil/predict.cc

```diff
@@ -70,100 +127,253 @@ inline int NextNodeCategorical(float fvalue, const std::vector<uint32_t>& matchi
   }
 }
 
-template <typename ThresholdType, typename LeafOutputType, typename DMatrixType,
-          typename OutputFunc>
-inline std::size_t PredictImplInner(const treelite::ModelImpl<ThresholdType, LeafOutputType>& model,
-                                    const DMatrixType* input, float* output,
-                                    const ThreadConfig& thread_config, bool pred_transform,
-                                    OutputFunc output_func) {
-  using TreeType = treelite::Tree<ThresholdType, LeafOutputType>;
-  const size_t num_row = input->GetNumRow();
-  const size_t num_col = input->GetNumCol();
-  std::vector<ThresholdType> row(num_col);
-  const treelite::TaskParam task_param = model.task_param;
-  std::vector<float> sum_tloc(task_param.num_class * thread_config.nthread);
-  std::vector<float> sum_tot(task_param.num_class);
+constexpr std::size_t kBlockOfRowsSize = 64;
```

Comment:
Be aware of data sparsity and the number of columns when specifying this parameter, it can take up a lot of memory considering that it's coupled with the number of threads.

Reply:
Noted.